File: wear/src/main/java/info/nightscout/androidaps/interaction/utils/SafeParse.java
package info.nightscout.androidaps.interaction.utils;

/**
 * Created by mike on 23.06.2016.
 */
public class SafeParse {
    public static Double stringToDouble(String input) {
        Double result = 0d;
        input = input.replace(",", ".");
        try {
            result = Double.parseDouble(input);
        } catch (Exception e) {
        }
        return result;
    }

    public static Integer stringToInt(String input) {
        Integer result = 0;
        input = input.replace(",", ".");
        try {
            result = Integer.parseInt(input);
        } catch (Exception e) {
        }
        return result;
    }

   public static Long stringToLong(String input) {
       Long result = 0L;
       input = input.replace(",", ".");
       try {
           result = Long.parseLong(input);
       } catch (Exception e) {
       }
       return result;
   }
   
   public static Float stringToFloat(String input) {
       Float result = 0f;
       input = input.replace(",", ".");
       try {
           result = Float.valueOf(input);
       } catch (Exception e) {
       }
       return result;
    }
}

File: core/interfaces/src/main/java/info/nightscout/shared/SafeParse.kt
package info.nightscout.shared

object SafeParse {

    fun stringToDouble(inputString: String?, defaultValue: Double = 0.0): Double {
        var input = inputString ?: return defaultValue
        var result = defaultValue
        input = input.replace(",", ".")
        input = input.replace("−", "-")
        if (input == "") return defaultValue
        try {
            result = input.toDouble()
        } catch (ignored: Exception) {
            // fail over to default
        }
        return result
    }

    fun stringToFloat(inputString: String?, defaultValue: Float = 0f): Float {
        var input = inputString ?: return defaultValue
        var result = defaultValue
        input = input.replace(",", ".")
        input = input.replace("−", "-")
        if (input == "") return defaultValue
        try {
            result = input.toFloat()
        } catch (ignored: Exception) {
            // fail over to default
        }
        return result
    }

    fun stringToInt(inputString: String?, defaultValue: Int = 0): Int {
        var input = inputString ?: return defaultValue
        var result = defaultValue
        input = input.replace(",", ".")
        input = input.replace("−", "-")
        if (input == "") return defaultValue
        try {
            result = input.toInt()
        } catch (ignored: Exception) {
            // fail over to default
        }
        return result
    }

    fun stringToLong(inputString: String?, defaultValue: Long = 0L): Long {
        var input = inputString ?: return defaultValue
        var result = defaultValue
        input = input.replace(",", ".")
        input = input.replace("−", "-")
        if (input == "") return defaultValue
        try {
            result = input.toLong()
        } catch (ignored: Exception) {
            // fail over to default
        }
        return result
    }
}